### PR TITLE
add: support for messageId

### LINKED ISSUE #458: Add support for messageId
**Is your feature request related to a problem? Please describe.**
Currently, there is no support for a unique identifier for messages. My use case is to use `messageId` for validation.

**Describe the solution you'd like**
We have [operationId](https://asyncapi.io/docs/specifications/2.0.0#operationTraitObject). `messageId` would be the exact same field for messages.

**Describe alternatives you've considered**
For now, I'm relying on the `name` field and adding extra checks on it. But it's not required to be unique by specs.

**Additional context**
Related slack conv: https://asyncapi.slack.com/archives/C34F2JV0U/p1604399108296600
Another previous conv about adding messageId: https://github.com/asyncapi/asyncapi/issues/94#issuecomment-576433439
Related request from tooling: https://github.com/WaleedAshraf/asyncapi-validator/issues/25

### PR DESCRIPTION
**Pull request by WaleedAshraf**
fixes #458

### CODE REVIEW

File: versions/2.0.0/asyncapi.md [hunk 1]
```diff
@@ -993,6 +993,7 @@ Field Name | Type | Description
 ---|:---:|---
 <a name="messageObjectHeaders"></a>headers | [Schema Object](#schemaObject) &#124; [Reference Object](#referenceObject) | Schema definition of the application headers. Schema MUST be of type "object". It **MUST NOT** define the protocol headers.
 <a name="messageObjectPayload"></a>payload | `any` | Definition of the message payload. It can be of any type but defaults to [Schema object](#schemaObject).
+<a name="messageObjectMessageId"></a>messageId | `string` | Unique string used to identify the message. The id MUST be unique among all messages described in the API. The messageId value is **case-sensitive**. Tools and libraries MAY use the messageId to uniquely identify an message, therefore, it is RECOMMENDED to follow common programming naming conventions.
```
Comment (fmvilas) [inline versions/2.0.0/asyncapi.md:996]:
```suggestion
<a name="messageObjectMessageId"></a>messageId | `string` | Unique string used to identify the message. The id MUST be unique among all messages described in the API. The messageId value is **case-sensitive**. Tools and libraries MAY use the messageId to uniquely identify a message, therefore, it is RECOMMENDED to follow common programming naming conventions.
```

**Review by fmvilas — COMMENTED** [review]
Just left a minor typo fix. Looking great so far for me. Thanks, @WaleedAshraf!

**Review by fmvilas — APPROVED** [review]

**Comment by WaleedAshraf** [comment]
related PR: https://github.com/asyncapi/asyncapi-node/pull/24

**Comment by derberg** [comment]
@WaleedAshraf please adjust the PR, schema is not here in this repo anymore, and spec changed location internally

**Comment by WaleedAshraf** [comment]
@derberg I'll close this for now. Once we know when it can be merge, I'll make new PRs in the respective repo. 👍

**Pull request closed without merging**